commands/system/help.js

```diff
@@ -0,0 +1,326 @@
+      extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
+      category: "System",
+      usage: "help <category/command/setting>",
+      aliases: ["h", "halp", "commands"],
+    });
+  }
+
+  async run(message, [type], level) {
+    var page = 1;
+    var n = 0;
+    var finalpage;
+    var num;
+
+    const embed = new MessageEmbed()
+      .setTimestamp()
+      .setColor(message.guild.me.roles.highest.color || 5198940)
+      .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL());
+
+    let currentCategory = "";
+    const sorted = this.client.commands.sort((p, c) => (p.help.cat > c.help.cat ? 1 : p.help.name > c.help.name && p.help.cat === c.help.cat ? 1 : -1));
+    if (!type) {
+      const description = `Command category list\n\nUse \`${message.settings.prefix}help <category>\` to find commands for a specific category`;
+      const output = sorted.filter(c => !(level < 10 && c.help.category === "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map((c) => {
+        const cat = c.help.cat;
+        if (currentCategory !== cat && !type) {
+          currentCategory = cat;
+          return `\n\`${message.settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands`;
+        }
+      }).join("");
+      embed.setDescription(description)
+        .addField("Categories", output);
+    } else {
+      sorted.forEach((c) => {
+        if (c.help.cat.toLowerCase() === type.toLowerCase()) {
+          n = n + 1;
+        }
+      });
+
+      var output = "";
+      num = 0;
+      const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
+      for (const c of sorted.values()) {
+        if (c.help.cat.toLowerCase() === type.toLowerCase()) {
+          if (c.help.cat === "NSFW" && !message.channel.nsfw) return;
+          if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
+            if (level < this.client.levelCache[c.conf.permLevel]) return;
+            output = output + `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 80 ? `${c.help.description.slice(0, 80)}...` : c.help.description}`;
+          }
+          num = num + 1;
+        }
+      }
+
+      if (num) {
+        finalpage = Math.ceil(num / perpage);
+        embed.setTitle(`Page ${page}/${Math.ceil(num / perpage)}`)
+          .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\`\n\n${num > 10 && pg === 1 ? `To view more commands do\` ${message.settings.prefix}help <category> 2\`` : ""}`)
+          .addField("Commands", output);
+      }
+    }
+
+    if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
```

Comment:
This could be replaced with get and find to avoid checking things twice. And use `this.client.aliases` instead.

Reply:
Again, yukine

Reply:
still not me